crates/core/src/operations/load_cdf.rs

```diff
@@ -0,0 +1,471 @@
+    /// Which iterates through versions of the delta table collects the relevant actions / commit info and returns those
+    /// groupings for later use. The scala implementation has a lot more edge case handling and read schema checking (and just error checking in general)
+    /// than I have right now. I plan to extend the checks once we have a stable state of the initial implementation.
+    async fn determine_files_to_read(
+        &self,
+    ) -> DeltaResult<(
+        Vec<CdcDataSpec<AddCDCFile>>,
+        Vec<CdcDataSpec<Add>>,
+        Vec<CdcDataSpec<Remove>>,
+    )> {
+        let start = self.starting_version;
+        let end = self
+            .ending_version
+            .unwrap_or(self.log_store.get_latest_version(start).await?);
+
+        let starting_timestamp = self.starting_timestamp.unwrap_or(DateTime::UNIX_EPOCH);
+        let ending_timestamp = self
+            .ending_timestamp
+            .unwrap_or(DateTime::from(SystemTime::now()));
+
+        log::debug!(
+            "starting timestamp = {:?}, ending timestamp = {:?}",
+            &starting_timestamp,
+            &ending_timestamp
+        );
+        log::debug!("starting version = {}, ending version = {:?}", start, end);
+
+        let mut change_files = vec![];
+        let mut add_files = vec![];
+        let mut remove_files = vec![];
+
+        for version in start..=end {
+            let snapshot_bytes = self
+                .log_store
+                .read_commit_entry(version)
+                .await?
+                .ok_or(DeltaTableError::InvalidVersion(version))?;
+            let version_actions = get_actions(version, snapshot_bytes).await?;
+
+            let mut ts = 0;
+            let mut cdc_actions = vec![];
+
+            if self.starting_timestamp.is_some() || self.ending_timestamp.is_some() {
+                let version_commit = version_actions
+                    .iter()
+                    .find(|a| matches!(a, Action::CommitInfo(_)));
+                if let Some(Action::CommitInfo(CommitInfo {
+                    timestamp: Some(t), ..
+                })) = version_commit
+                {
+                    if starting_timestamp.timestamp_millis() > *t
+                        || *t > ending_timestamp.timestamp_millis()
+                    {
+                        log::debug!("Version: {} skipped, due to commit timestamp", version);
+                        continue;
+                    }
+                }
+            }
+
+            for action in &version_actions {
+                match action {
+                    Action::Cdc(f) => cdc_actions.push(f.clone()),
+                    Action::Metadata(md) => {
+                        log::info!("Metadata: {:?}", &md);
+                        if let Some(Some(key)) = &md.configuration.get("delta.enableChangeDataFeed")
```

Comment:
Drive by comment, but the spark version seems to check that the starting version has this enabled, and then comments [here](https://github.com/delta-io/delta/blob/master/spark/src/main/scala/org/apache/spark/sql/delta/commands/cdc/CDCReader.scala#L524C1-L526C90) that the equivalent of this check is enough to ensure the total range has it enabled.

I don't see a check for the starting version anywhere in this PR though. Do you need to add one?